port/aix/omrsl.c

```diff
@@ -763,5 +763,63 @@ omrsl_startup(struct OMRPortLibrary *portLibrary)
 uintptr_t
 omrsl_get_libraries(struct OMRPortLibrary *portLibrary, OMRLibraryInfoCallback callback, void *userData)
 {
-	return OMRPORT_ERROR_NOT_SUPPORTED_ON_THIS_PLATFORM;
+	uintptr_t result = 0;
+	void *buffer = NULL;
+	struct ld_info *ldInfo = NULL;
+	int rc = 0;
+	unsigned int bufferSize = 65536;
+	for (;;) {
+		buffer = portLibrary->mem_allocate_memory(
+				portLibrary,
+				bufferSize,
+				OMR_GET_CALLSITE(),
+				OMRMEM_CATEGORY_PORT_LIBRARY);
+		if (NULL == buffer) {
+			portLibrary->error_set_last_error_with_message(
+					portLibrary,
+					-1,
+					"Failed to allocate memory for loadquery buffer.");
+			return (uintptr_t)(intptr_t)-1;
+		}
+		rc = loadquery(L_GETINFO, buffer, bufferSize);
+		if (-1 != rc) {
+			break;
+		}
+		if (ENOMEM != errno) {
+			int32_t portableError = portLibrary->error_last_error_number(portLibrary);
+			portLibrary->error_set_last_error_with_message(
+					portLibrary,
+					portableError,
+					"loadquery(L_GETINFO) failed.");
+			portLibrary->mem_free_memory(portLibrary, buffer);
+			return (uintptr_t)(intptr_t)portableError;
+		}
+		portLibrary->mem_free_memory(portLibrary, buffer);
+		bufferSize *= 2;
+	}
+	ldInfo = (struct ld_info *)buffer;
+	for(;;) {
```

Comment:
Missing space before `(`.

Earlier versions of this checked `if ('\0' != ldInfo->ldinfo_filename[0])`: why was that removed?

Omitting that check is inconsistent with the implementation in `unix/omrsl.c` which only reports entries where the path starts with `/`.

https://www.ibm.com/docs/hr/aix/7.2.0?topic=files-ldrh-file says:
> Nul-terminated path name followed by nul-terminated member name; member name is empty if not an archive.

I interpret that to mean that there may be two parts to the name (e.g. `libc.a(member.o)` would be represented as `"libc.a\0member.o"`); should we not report both parts to the caller (perhaps as `"libc.a(member.o)"`)?

Reply:
The space between `for` and `(` is still missing.